doc/dev/index.rst

```diff
@@ -265,6 +329,82 @@ the `Git Commit Good Practice`_ article at the `OpenStack Project Wiki`_.
 .. _`Git Commit Good Practice`: https://wiki.openstack.org/wiki/GitCommitMessages
 .. _`OpenStack Project Wiki`: https://wiki.openstack.org/wiki/Main_Page
 
+Once your pull request is opened, update the `Issue tracker` by adding an
+update to the bug. The update can be as simple as:
+
+.. code:
+
+    *PR*: https://github.com/ceph/ceph/pull/$NUMBER_OF_YOUR_PULL_REQUEST
+
+Automated PR validation
+-----------------------
+
+When your PR hits GitHub, the Ceph project's Continual Integration (CI)
+infrastructure will test it automatically. At the time of this writing, the
+automated CI testing included a test to check that the commits in the PR
+are properly signed (see `Submitting patches`_) and a `make check` test.
+
+The latter, `make check`, builds the PR and runs it through a battery of
+unit tests. When these complete, the result will be shown on GitHub in the
+pull request itself.
+
+You can also run `make check` locally before you open a PR. The easiest way
+to do this is to execute the following command from the top of the source tree:
+
+.. code:
+
+    $ ./run-make-check.sh
+
+You will need a minimum of 8GB of RAM and 32GB of free disk space for this
+command to complete successfully. Depending on your hardware, it can take a
+very long time to complete, but it's worth the wait.
+
+For more information on unit tests, read the `Testing`_ chapter.
+
+ceph-qa-suite
+-------------
+
+Since Ceph is a complex beast, it may also be necessary to test your fix to
+see how it behaves on real clusters running either on real or virtual
+hardware. Tests designed for this purpose live in the `ceph-qa-suite
+repository <https://github.com/ceph/ceph-qa-suite>` and are run via the
+`teuthology framework`_.
+
+If you have access to an OpenStack tenant, you can run the integration
+tests yourself using `teuthology's OpenStack backend`_. The Ceph community
+also operates the `Sepia lab`_ where the integration tests can be run on
+real hardware.
+
+Integration tests are discussed in detail in the `Testing`_ chapter.
+
+Code review
+-----------
+
+Once your bugfix has been thoroughly tested, or even during this process,
+it will be subjected to code review by other developers. This typically
+takes the form of correspondence in the PR itself, but can be supplemented
+by discussions on `IRC`_ and the `ceph-devel` `Mailing list`_.
+
+Amending your PR
+----------------
+
+While your PR is going through `Testing`_ and `Code review`_, you can
+modify it at any time by editing files locally in your branch (`fix_1` in
+our example), committing the changes, and "force-pushing" the "WIP branch".
```

Comment:
instead of

    , and "force-pushing" the "WIP branch".

When such changes modify the git history (because of a git rebase or git commit --amend), you will need to force push your branch with:

    git push --force origin fix_1